### PR TITLE
Use Nix and Cabal instead of Stack for building

### PR DESCRIPTION
**Pull request by mrkkrp**

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        { "markkarpov-com" = super.callCabal2nix "markkarpov-com" (pkgs.lib.sourceByRegex ./. sourceRegex) {}; } ;
+    };
+  op =
+    if pkgs.lib.inNixShell
+    then haskellPackages.shellFor
+      { packages = (ps: [ ps.markkarpov-com ]);
+        buildInputs = [ pkgs.cabal-install ];
+      }
+    else haskellPackages.markkarpov-com;
+in op
```
Comment (srid) [inline default.nix]: Is there a reason `developPackage` didn't work out?

Comment (mrkkrp) [reply default.nix]: It doesn't work for `nix-build` only for `nix-shell`. Basically `shellFor` ~ `developPackage`, but those are meant to be used in shells, not in `nix-build`.

Comment (srid) [reply default.nix]: `nix-build` works for me with `developPackage`:

```
srid@thebeast /t/juncture> cat default.nix
let
  pkgs = import <nixpkgs> { };
  H = pkgs.haskellPackages;
  HL = pkgs.haskell.lib;
in H.developPackage {
  root = ./.;
  overrides = self: super: {
    # 0.4 version is not released yet.
    algebraic-graphs = HL.dontCheck (H.callCabal2nix "algebraic-graphs" (builtins.fetchTarball {
      url = "https://github.com/snowleopard/alga/archive/1754312.tar.gz";
      sha256 = "09gp4vgslcr9r5w8h5jm1la0c5fnzxmhiv530rq7i16sisfrj4rr";
    }) {});
  };
}
srid@thebeast /t/juncture> nix-build
these derivations will be built:
[...]
/nix/store/kqxvnhw72r5gg8z82mmmjcshazgaw7fx-juncture-0.1.0.0
srid@thebeast /t/juncture> tree result
result
└── bin
    └── juncture

1 directory, 1 file
srid@thebeast /t/juncture>
```

Comment (mrkkrp) [reply default.nix]: Actually it looks like you're right, it should work with `developPackage` too, the issue I had seems to be unrelated. I'll try with `developPackage` again.

Comment (mrkkrp) [reply default.nix]: I don't see how to filter out some source files from `root` directory with `developPackage` though.

Comment (srid) [reply default.nix]: Off the hand I don't know how to do that, but why not put the source files under a sub-directory (being its own cabal project)?

Comment (mrkkrp) [reply default.nix]: That's not handy as I do a lot of various stuff locally and I want to be able to run `niix-build` without including tons of unrelated files as inputs of the derivation.

**Pull request merged**